cpp/include/cudf_conversion.hpp

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_view.hpp>
+
+#include <graph.hpp>
+#include <utilities/error_utils.h>
+
+namespace cugraph {
+namespace experimental {
+
+namespace detail {
+
+void check_edge_list_columns(cudf::column_view const &src_indices,
+                             cudf::column_view const &dst_indices) {
+  CUGRAPH_EXPECTS( src_indices.null_count() == 0 , "Input column has non-zero null count");
+  CUGRAPH_EXPECTS( dst_indices.null_count() == 0 , "Input column has non-zero null count");
+  CUGRAPH_EXPECTS( src_indices.type() == dst_indices.type() , "Unsupported data type" );
+  CUGRAPH_EXPECTS( src_indices.type().id() == cudf::experimental::type_to_id<int32_t>() , "Unsupported data type" );
+  CUGRAPH_EXPECTS( src_indices.size() == dst_indices.size() , "Source and Destination must have same number of rows" );
+  CUGRAPH_EXPECTS( src_indices.size() > 0 , "No edges");
+}
+
+void check_adj_list_columns(cudf::column_view const &indices,
+                            cudf::column_view const &offsets) {
+  CUGRAPH_EXPECTS( offsets.null_count() == 0 , "Input column has non-zero null count");
+  CUGRAPH_EXPECTS( indices.null_count() == 0 , "Input column has non-zero null count");
+  CUGRAPH_EXPECTS( offsets.type() == indices.type() , "Unsupported data type" );
+  CUGRAPH_EXPECTS( offsets.type().id() == cudf::experimental::type_to_id<int32_t>() , "Unsupported data type" );
+  CUGRAPH_EXPECTS( offsets.size() > 0 , "Offsets column is empty");
+}
+
+}
+
+/**
+ * @Synopsis   Wrap existing cudf columns representing an edge list in a Graph.
+ *             cuGRAPH does not own the memory used to represent this graph. This
+ *             function does not allocate memory.
+ *
+ * @Param[in]  source_indices        This column_view of size E (number of edges) contains the index of the source for each edge.
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in]  destination_indices   This column_view of size E (number of edges) contains the index of the destination for each edge.
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in]  number_of_vertices    The number of vertices in the graph
+ * @Param[in]  number_of_edges       The number of edges in the graph
+ *
+ * @Returns    GraphCOO object
```

Comment:
Take this with a grain of salt, but IIRC, @returns require return type before explanation.

e.g.
@returns GraphCOO<VT,WT> GraphCOO object